docs/reference/vectors/vector-functions.asciidoc

```diff
@@ -195,3 +205,50 @@ You can check if a document has a value for the field `my_vector` by
 "source": "doc['my_vector'].size() == 0 ? 0 : cosineSimilarity(params.queryVector, 'my_vector')"
 --------------------------------------------------
 // NOTCONSOLE
+
+The recommended way to access dense vectors is through `cosineSimilarity`,
+`dotProduct`, `l1norm` or `l2norm` functions. But for custom use cases,
+you can access dense vectors's values directly through the following functions:
+
+- `float[] getVectorValue()` – returns a vector's value as an array of floats
+
+- `float getVectorMagnitude()` – returns a vector's magnitude (available for
```

Comment:
@jtibshirani Thanks for the feedback. I was also thinking the same – to implement a slower version of `getVectorMagnitude`, but this would require decoding the whole vector. If a user is already using `vectorValue` in their script and decoding a vector, using `magnitude` would mean decoding this vector the second time.  So it would be faster for a user to implement `magnitude` function themselves. WDYT?

Reply:
To me it's the right trade-off for a simple API. First, it will only be slow for vectors indexed before 7.5, which was before they were even GA. It also seems okay that it's slow, users can easily work around it. Maybe we could just write a short note in the docs about the pre-7.5 behavior so that users are aware.